crates/js-sys/src/lib.rs

```diff
@@ -3641,6 +3641,31 @@ impl JsString {
     ) -> impl ExactSizeIterator<Item = u16> + DoubleEndedIterator<Item = u16> + 'a {
         (0..self.length()).map(move |i| self.char_code_at(i) as u16)
     }
+
+    /// If this string consists of a single UTF-8 codepoint, then this function
+    /// returns a Rust representation of that character without any allocations.
+    ///
+    /// If this JS value is not a valid UTF-8 or consists of more than a single
+    /// codepoint, then this returns `None`.
```

Comment:
I think it would be useful to clarify to the user what exactly a "UTF-8 codepoint" is, since it's easy to confuse "code point" with "scalar value" or "character" or "char code".

Also, "UTF-8 codepoint" is technically incorrect, since UTF-8 is an encoding, but code points transcend encodings.

So I would rename it to "Unicode code point", and I would also add a small section describing what it is:

> A JavaScript string that contains multiple characters might actually be a single Unicode code point. For example, `"\uD801\uDC37"` is the single Unicode code point `U+10437` (which is the same as the Rust character `'𐐷'`).

Reply:
> since it's easy to confuse "code point" with "scalar value" or "character" or "char code"

I'd normally expect people to be familiar with "code point" vs "code unit" / "char code" distinction if they work with JS strings, but I guess doesn't harm adding an explanation.

Reply:
Done, although I've reworded it since "character" is an ambiguous term, especially in this context.

Reply:
I think most programmers in general (whether JavaScript or not) have no idea how Unicode works, and frequently get the terms mixed up.

Most programmers tend to just assume everything is ASCII.

Reply:
Nit: "corresponds to a character '𐐷'." -> "corresponds to the character '𐐷'."